subprojects/testing-base/src/test/groovy/org/gradle/api/internal/tasks/testing/junit/result/JUnitXmlResultWriterSpec.groovy

```diff
@@ -248,4 +248,50 @@ class JUnitXmlResultWriterSpec extends Specification {
         generator.write(result, text)
         return text.toString("UTF-8").replace(SystemProperties.instance.lineSeparator, "\n")
     }
+
+    @Issue("https://github.com/gradle/gradle/issues/23229")
+    def "omit system-out section"() {
+        given:
+        options = new JUnitXmlResultOptions(true, false, true, false)
+
+
+        TestClassResult result = new TestClassResult(1, "com.foo.FooTest", startTime)
+        result.add(new TestMethodResult(1, "some test").completed(new DefaultTestResult(SUCCESS, startTime + 100, startTime + 300, 1, 1, 0, emptyList())))
+        _ * provider.writeAllOutput(_, _, _)
+
+        when:
+        def xml = getXml(result)
+
+        then:
+        xml == """<?xml version="1.0" encoding="UTF-8"?>
+<testsuite name="com.foo.FooTest" tests="1" skipped="0" failures="0" errors="0" timestamp="2012-11-19T17:09:28" hostname="localhost" time="0.3">
+  <properties/>
+  <testcase name="some test" classname="com.foo.FooTest" time="0.2"/>
+  <system-err><![CDATA[]]></system-err>
```

Comment:
❓ Are the `system-err` (and `system-out`) elements required in the JUnit xml schema?  Can they be omitted entirely instead of added but containing empty text?

Reply:
this is very good question, I'll perform some additional checks